src/Enum.php

```diff
@@ -175,13 +185,27 @@ public static function toArray()
      * @param $value
      * @psalm-param mixed $value
      * @psalm-pure
+     * @psalm-assert-if-true T $value
      * @return bool
      */
     public static function isValid($value)
     {
         return \in_array($value, static::toArray(), true);
     }
 
+    /**
+     * Asserts valid enum value
+     *
+     * @psalm-pure
+     * @psalm-assert T $value
+     */
+    public static function assertValidValue($value): void
```

Comment:
I propose to not extend the API also with this static method and have it private.
I wanted to change how it behaves here: https://github.com/myclabs/php-enum/pull/136#discussion_r575875238

Reply:
@drealecs the use-case for this method is to assert on input values from unknown contexts: the API is relatively minimal, and doesn't need to rely on the optimized structure you worked on in #136.

Reply:
I understand what you say. Yes, it can be useful as a public API as well to just validate without creating the instances.
It would have help a bit in making sure `is_array` or `array_search` are not called in constructor more than once when object is created,
`is_array` is already called twice when `from()` is used and I was thinking that should also be optimized.

Reply:
And this endpoint can be made completely independent from the ctor in the patch doing the perf optimization, no?

Reply:
Yes, it can.